opm/input/eclipse/Schedule/Well/WellMatcher.hpp

```diff
@@ -35,21 +35,127 @@ namespace Opm {
 class WellMatcher
 {
 public:
+    /// Default constructor.
     WellMatcher() = default;
+
+    /// Constructor.
+    ///
+    /// Assumes ownership of the NameOrder object.
+    ///
+    /// \param[in] well_order Facility for ordering a list of well names
+    /// according to the associate insertion index.
     explicit WellMatcher(NameOrder&& well_order);
+
+    /// Constructor.
+    ///
+    /// Does not assume ownership of the NameOrder object which, therefore,
+    /// must outlive the WellMatcher object.  This is an optimisation for
+    /// the common case of short-lived WellMatcher objects with backing
+    /// store in a containing Schedule object.
+    ///
+    /// \param[in] well_order Facility for ordering a list of well names
+    /// according to the associate insertion index.
     explicit WellMatcher(const NameOrder* well_order);
+
+    /// Constructor.
+    ///
+    /// \param[in] wells List of wells for which to establish a matcher.
     explicit WellMatcher(std::initializer_list<std::string> wells);
+
+    /// Constructor.
+    ///
+    /// \param[in] wells List of wells for which to establish a matcher.
     explicit WellMatcher(const std::vector<std::string>& wells);
+
+    /// Constructor.
+    ///
+    /// Does not assume ownership of either the NameOrder or the
```

Comment:
neither/nor

Reply:
> neither/nor

No, I don't think so because that'd be a double negation.  If it were "Assumes ownership of...", then it'd be correct to have neither/nor here, but as written I think it's better to have either/or here.

Reply:
yeah you are correct, missed the first negation.